lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -389,6 +411,47 @@ instance FromJSON (ApiT WalletState) where
 instance ToJSON (ApiT WalletState) where
     toJSON = genericToJSON walletStateOptions . getApiT
 
+instance ToJSON ApiUtxoStatistics where
+    toJSON (ApiUtxoStatistics totalStake theScale bars) =
+        let
+            histogramObject =
+                Aeson.Object . HMS.fromList . map extractBarKey
+
+            extractBarKey (HistogramBarCount bound stake) =
+                showTextData bound .= stake
+        in
+            object
+                [ "total" .= totalStake
+                , "boundType" .= theScale
+                , "distribution" .= histogramObject bars
+                ]
```

Comment:
Any ways to do this generically :thinking: ? Haven't really got my head around it here, but types looks simple enough to rely on generic encoding in this case, na?

Reply:
probably yes, but then I need to deliver ToJson and FromJson instances for `[HistogramBar]` :thinking

Reply:
_not necessarily_  I mean, we do use a different type in the API, right? So, the `ApiUtxoStatistics` could directly use a `Map _ _` instead of a plain list (to make it play nicely with the generic encoding). And then, it's a matter of doing a `Map.fromList ...` or similar in the API handler when we convert from the core type to the API type. 

I think that could work pretty nicely without too much effort?

Reply:
very reasonable, implemented and behaves as you expected :-)